server/src/main/java/org/opensearch/common/lucene/Lucene.java

```diff
@@ -162,6 +163,23 @@ public static SegmentInfos readSegmentInfos(Directory directory) throws IOExcept
         return SegmentInfos.readLatestCommit(directory);
     }
 
+    /**
+     * A variant of {@link #readSegmentInfos(Directory)} that supports reading indices written by
+     * older major versions of Lucene. The underlying implementation is a workaround since the
+     * "expert" readLatestCommit API is currently package-private in Lucene. First, all commits in
+     * the given {@link Directory} are listed - this result includes older Lucene commits. Then,
+     * the latest index commit is opened via {@link DirectoryReader} by including a minimum supported
+     * Lucene major version - this is based on the minimum index compatiblity version of OpenSearch.
+     */
+    public static SegmentInfos readAnySegmentInfos(Directory directory) throws IOException {
```

Comment:
This is set up to be a utility method, but the ability to read "any" infos depends on the minimumIndexCompatibilityVersion.  Could we isolate this logic to SS only inside of ReadOnlyEngine?